pkg/master/controller.go

```diff
@@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: ports,
 			// maintained by this code, not by the pod selector
 			Selector:        nil,
 			ClusterIP:       serviceIP.String(),
```

Comment:
IFF we are willing to take a dep from kubelet -> kube-proxy, which we do
not currently do

On Fri, Sep 25, 2015 at 10:22 AM, jay vyas <notifications@github.com> wrote:

> In pkg/master/controller.go
> <https://github.com/kubernetes/kubernetes/pull/13978#discussion_r40453559>
> :
>
> > @@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
> >  			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
> >  		},
> >  		Spec: api.ServiceSpec{
> > -			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
> > +			Ports: ports,
> >  			// maintained by this code, not by the pod selector
> >  			Selector:        nil,
> >  			ClusterIP:       serviceIP.String(),
>
> I guess @thockin <https://github.com/thockin> is saying that kubelets can
> access the service using whatever clusterIP is auto-allocated to it....
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13978/files#r40453559>.
>